nx/test/nx/lin_alg_test.exs

```diff
@@ -260,16 +260,60 @@ defmodule Nx.LinAlgTest do
       assert_all_close(Nx.dot(q, r), t)
     end
 
+    test "works with batches of matrices" do
+      t =
+        Nx.tensor([
+          [[1.0, 2.0, 3.0], [0.0, 4.0, 5.0], [0.0, 0.0, 6.0]],
+          [[1.0, 2.0, 3.0], [0.0, 10.0, 5.0], [0.0, 0.0, 20.0]]
+        ])
+
+      {q, r} = Nx.LinAlg.qr(t)
+
+      expected_q =
+        Nx.tensor([
+          [
+            [1.0, 0.0, 0.0],
+            [0.0, 1.0, 0.0],
+            [0.0, 0.0, 1.0]
+          ],
+          [
+            [1.0, 0.0, 0.0],
+            [0.0, 1.0, 0.0],
+            [0.0, 0.0, 1.0]
+          ]
+        ])
+
+      assert_all_close(q, expected_q, atol: 1.0e-10)
+
+      expected_r =
+        Nx.tensor([
+          [
+            [1.0, 2.0, 3.0],
+            [0.0, 4.0, 5.0],
+            [0.0, 0.0, 6.0]
+          ],
+          [
+            [1.0, 2.0, 3.0],
+            [0.0, 10.0, 5.0],
+            [0.0, 0.0, 20.0]
+          ]
+        ])
+
+      assert_all_close(r, expected_r, atol: 1.0e-10)
+
+      assert_all_close(Nx.dot(q, [2], [0], r, [1], [0]), t, atol: 1.0e-10)
+    end
+
     test "property" do
```

Comment:
Reminder for myself to check if there's a property test in torchx and exla as well to change

Reply:
I think we'll also have to do some work to get the grad for the batched version to work because it uses Nx.LinAlg.invert